library/src/main/java/com/datatorrent/lib/io/jms/JMSBase.java

```diff
@@ -76,18 +76,24 @@
   private transient Session session;
   private transient Destination destination;
 
+  private ConnectionFactoryBuilder connectionFactoryBuilder;
   private String connectionFactoryClass;
   private Map<String, String> connectionFactoryProperties = Maps.newHashMap();
   private String ackMode = "CLIENT_ACKNOWLEDGE";
-  private String clientId = "TestClient";
-  private String subject = "TEST.FOO";
+  private String clientId;
+  private String subject;
```

Comment:
Is subject a parameter that user has to set? In that case add a constraint validation.